tls/s2n_signature_algorithms.c

```diff
@@ -13,11 +13,32 @@
  * permissions and limitations under the License.
  */
 
+#include "crypto/s2n_fips.h"
+#include "error/s2n_errno.h"
 #include "tls/s2n_cipher_suites.h"
 #include "tls/s2n_tls_digest_preferences.h"
 #include "tls/s2n_signature_algorithms.h"
+#include "utils/s2n_safety.h"
 
-int s2n_set_signature_hash_pair_from_preference_list(struct s2n_connection *conn, uint8_t sig_hash_algs[TLS_SIGNATURE_ALGORITHM_COUNT][TLS_HASH_ALGORITHM_COUNT], 
+static int s2n_sig_hash_algs_pairs_set(struct s2n_sig_hash_alg_pairs *sig_hash_algs, uint8_t sig_alg, uint8_t hash_alg)
+{
+    S2N_ERROR_IF(hash_alg >= TLS_HASH_ALGORITHM_COUNT, S2N_ERR_HASH_INVALID_ALGORITHM);
```

Comment:
If the `hash_alg` value is coming from peer's `signature_algorithm` extension, shouldn't we skip an unknown value rather than throwing an error? The peer could be sending a code point that s2n has not yet implemented. If there is *no* overlap between the peer's set and ours we will error later during cipher suite/cert selection.

Reply:
Will follow up with a patch.

Reply:
Actually, I this is okay because we aren't checking the return value of `s2n_sig_hash_alg_pairs_set` in `s2n_recv_supported_signature_algorithms`. I'll validate with a little unit test.